### PR TITLE
Dont set the active sessions if using ignoreMissingSessions

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

File: sakora-csv-impl/impl/src/java/net/unicon/sakora/impl/csv/CsvAcademicSessionHandler.java [hunk 1]
```diff
@@ -122,8 +122,12 @@ protected void processInternal(CsvSyncContext context) {
              */
             log.error("SakoraCSV has no current academic sessions to process and session processing is on, this is an invalid state as it would cause all data (courses, memberships, etc.) to be disabled or removed, please check your sessions.csv file");
         }
-        // set the current academic sessions according to the incoming sessions
-		cmAdmin.setCurrentAcademicSessions(currentSessions);
+
+		if (!currentSessions.isEmpty() && !commonHandlerService.ignoreMissingSessions()) {
+			// set the current academic sessions according to the incoming sessions
+			cmAdmin.setCurrentAcademicSessions(currentSessions);
```
Comment (bjones86) [inline sakora-csv-impl/impl/src/java/net/unicon/sakora/impl/csv/CsvAcademicSessionHandler.java:128]: What action does this method perform that this change would prevent?

File: sakora-csv-impl/impl/src/java/net/unicon/sakora/impl/csv/CsvAcademicSessionHandler.java [hunk 2]
```diff
@@ -122,8 +122,12 @@ protected void processInternal(CsvSyncContext context) {
...
+		}
+
 		commonHandlerService.setCurrentAcademicSessions( currentSessions.toArray(new String[currentSessions.size()]) );
```
Comment (bjones86) [inline sakora-csv-impl/impl/src/java/net/unicon/sakora/impl/csv/CsvAcademicSessionHandler.java:131]: Does this call do the same thing? The name is identical, should this also be inside the introduced conditional?

**Review by bjones86 — COMMENTED** [review]
@ottenhoff, any info on this change? What was the problem that necessitated this? We run with a sessions file and have `ignoreMissingSessions` set to `true`, so I'd like to understand what the problem is and how it will affect us.

**Comment by ottenhoff** [comment]
I believe the issue is if you run two totally different sets of CSV files corresponding to two campuses. Then the sessions file for A will set the currentSessions which indicates which sessions are available in the Worksite Setup UI

File: sakora-csv-impl/impl/src/java/net/unicon/sakora/impl/csv/CsvAcademicSessionHandler.java (hunk 2 above)
Comment (ottenhoff) [reply sakora-csv-impl/impl/src/java/net/unicon/sakora/impl/csv/CsvAcademicSessionHandler.java:131]: No. I think this is what sets the sessions being run currently in sakora-csv

**Comment by bjones86** [comment]
Hmm... I'm still a little confused. Is an institution running sakora twice and using `ignoreMissingSessions` to have it treat them as exclusive updates?

Here's some context: we use `ignoreMissingSessions` to freeze old enrolments so people aren't mass unenrolled from old sites, and also so we don't have to process years worth of old enrolments that never get updated. I'm unsure how this change would affect our use case: would it prevent us from ever updating our _active_ sessions?

**Comment by ottenhoff** [comment]
> Hmm... I'm still a little confused. Is an institution running sakora twice and using ignoreMissingSessions to have it treat them as exclusive updates?

Yes

>  would it prevent us from ever updating our active sessions?

Yes, sakora-csv wouldn't touch your active sessions

**Comment by bjones86** [comment]
I see, thanks for the explanation. For us this change is undesirable. Is it possible to put this behaviour behind a new property?

**Comment by ottenhoff** [comment]
Got it. Closing for now

**Pull request closed without merging**